core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -487,4 +487,54 @@ public void testIncludedPartitionSummaryLimit() {
         table.currentSnapshot().summary().get(SnapshotSummary.CHANGED_PARTITION_COUNT_PROP);
     Assert.assertEquals("Should set changed partition count", "2", changedPartitions);
   }
+
+  @Test
+  public void testAppendToExistingBranch() {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    table.manageSnapshots().createBranch("branch", table.currentSnapshot().snapshotId()).commit();
+    table.newFastAppend().appendFile(FILE_B).toBranch("branch").commit();
```

Comment:
Just realized I just hardcoded branch throughout the tests, probably cleaner to just define a variable in the tests and use that.

Reply:
I think this is okay.